experimental/JuLie/quantum_analogs.jl

```diff
@@ -0,0 +1,295 @@
+
+# Examples
+```jldoctest
+julia> quantum_integer(3)
+q^2 + q + 1
+
+julia> quantum_integer(-3)
+-q^-1 - q^-2 - q^-3
+
+julia> quantum_integer(3,2)
+7
+
+julia> quantum_integer(-3,2)
+-7//8
+
+julia> K,i = cyclotomic_field(4, "i");
+
+julia> quantum_integer(3, i)
+i
+```
+
+# References
+1. [Con00](@cite)
+2. [KC02](@cite)
+"""
+function quantum_integer(n::IntegerUnion, q::RingElem)
+
+  R = parent(q)
+	if isone(q)
+		return R(n)
+	else
+		z = zero(R)
+		if n >= 0
+			for i = 0:n-1
+				z += q^i
+			end
+		else
+			for i = 0:-n-1
+				z -= q^(n+i)
+			end
+		end
+		return z
+	end
+end
+
+function quantum_integer(n::IntegerUnion, q::Integer)
+	if n >= 0 || q == 1 || q == -1
+		return quantum_integer(n,ZZ(q))
+	else
+		return quantum_integer(n,QQ(q))
+	end
+end
+
+function quantum_integer(n::IntegerUnion)
+	R,q = LaurentPolynomialRing(ZZ, "q")
+	return quantum_integer(n,q)
+end
+
+
+################################################################################
+# Quantum factorials
+################################################################################
+@doc Markdown.doc"""
+    quantum_factorial(n::IntegerUnion, q::RingElem)
+    quantum_factorial(n::IntegerUnion, q::Integer)
+    quantum_factorial(n::IntegerUnion)
+
+For a non-negative integer ``n`` and an element ``q`` of a ring ``R`` the **quantum
+factorial** ``[n]_q! ∈ R`` is defined as
+```math
+[n]_q! ≔ [1]_q ⋅ … ⋅ [n]_q ∈ R \;.
+```
+Note that for ``q=1`` we obtain
+```math
+[n]_1! = n! \quad \text{ for all } n ∈ ℤ \;,
+```
+hence the quantum factorial is a "deformation" of the usual factorial.
+
+# Functions
+
+The functions `quantum_factorial` work analogously to [`quantum_integer`](@ref).
+
+# Examples
+```jldoctest
+julia> quantum_factorial(3)
+q^3 + 2*q^2 + 2*q + 1
+
+julia> quantum_factorial(3,2)
+21
+
+julia> K,i = cyclotomic_field(4, "i");
+
+julia> quantum_factorial(3, i)
+i - 1
+```
+"""
+function quantum_factorial(n::IntegerUnion, q::RingElem)
+
+	n >= 0 || throw(ArgumentError("n ≥ 0 required"))
+
+	R = parent(q)
+	if isone(q)
+		return R(factorial(n))
+	else
+		z = one(R)
+		for i = 1:n
+			z *= quantum_integer(i,q)
+		end
+		return z
+	end
+end
+
+function quantum_factorial(n::IntegerUnion, q::Integer)
+	return quantum_factorial(n,ZZ(q))
+end
+
+function quantum_factorial(n::IntegerUnion)
+	R,q = LaurentPolynomialRing(ZZ, "q")
+	return quantum_factorial(n,q)
+end
+
+
+################################################################################
+# Quantum binomials
+################################################################################
+
+@doc Markdown.doc"""
+    quantum_binomial(n::IntegerUnion, q::RingElem)
+    quantum_binomial(n::IntegerUnion, k::IntegerUnion, q::Integer)
+    quantum_binomial(n::IntegerUnion, k::IntegerUnion)
+
+Let ``k`` be a non-negative integer and let ``n ∈ ℤ``. The **quantum binomial**
+``\begin{bmatrix} n \\ k \end{bmatrix}_𝐪 \in ℚ(𝐪)`` is defined as
```

Comment:
Perhaps use `\binom`, it is standard and the resulting TeX is more readable in the REPL. Also of course my comments from above also apply here (so I'd replace `𝐪` by `q`)
```suggestion
``\binom{n}{k}_𝐪 \in ℚ(𝐪)`` is defined as
```